### PR TITLE
Bug 1419054 - Don't create bufferBookmarksWithDeletionsView in v28 db migration

### PR DESCRIPTION
**Pull request by eoger**
We create that view in the v33 migration path anyway.

### CODE REVIEW

**Comment by rnewman** [comment]
Could you please hit the DB upgrade tests, and make them upgrade from something very early, like v7 — we already have `BrowserSchemaV7` in the tree — and make sure a stepwise upgrade (a) reproduces the bug, and (b) with your fix we can get all the way from `7` to `34`?

**Comment by eoger** [comment]
Done, thank you

File: Storage/SQL/BrowserSchema.swift [hunk 1]
```diff
@@ -1014,13 +1015,14 @@ open class BrowserSchema: Schema {
...
         }
 
-        if from < 20 && to >= 20 {
-            if !self.run(db, queries: [
-                "DROP VIEW IF EXISTS \(ViewBookmarksBufferOnMirror)",
-                self.bufferBookmarksView]) {
-                return false
-            }
-        }
+// That view is re-created later
```
Comment (rnewman) [inline Storage/SQL/BrowserSchema.swift]: Indenting on these first lines.

I was going to suggest writing a block comment instead, but I see commented-out migrations below, so fine.

File: StorageTests/TestSQLiteBookmarks.swift [hunk 2]
```diff
@@ -56,9 +208,26 @@ class TestSQLiteBookmarks: XCTestCase {
         self.remove("TSQLBtestLocalBookmarksModifications.db")
         self.remove("TSQLBtestApplyBufferUpdatedCompletionOp.db")
         self.remove("TSQLBtestApplyRecordsPendingDeletions.db")
+        self.remove("TSQLBtestDBUpgradeBeforeDateAdded.db")
         super.tearDown()
     }
 
+    func testDateAddedMigration() {
+        let schema = BrowserDBV15()
```
Comment (rnewman) [inline StorageTests/TestSQLiteBookmarks.swift]: Does it work if this is `V7`?

**Pull request merged**